### ISSUE TITLE
No option to enable soft delete for resource azurerm_data_protection_backup_vault

### ISSUE THREAD

**Issue by Peter0401** [hashicorp/terraform-provider-azurerm#23369]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.7

### AzureRM Provider Version

3.74.0

### Affected Resource(s)/Data Source(s)

azurerm_data_protection_backup_vault

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "3.74.0"
    }
  }
}

provider "azurerm" {
  features {}
  skip_provider_registration = true
}

data "azurerm_resource_group" "test_rg_01" {
  name = "tf-provider-test-5000"
}

resource "azurerm_data_protection_backup_vault" "test_bv_01" {
  name                = "test-backup-vault-01"
  resource_group_name = data.azurerm_resource_group.test_rg_01.name
  location            = data.azurerm_resource_group.test_rg_01.location
  datastore_type      = "VaultStore"
  redundancy          = "GeoRedundant"
}
```

### Debug Output/Panic Output

```shell
bash-3.2$ terraform --version
Terraform v1.5.7
on darwin_arm64
bash-3.2$ terraform init

Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "3.74.0"...
- Installing hashicorp/azurerm v3.74.0...
- Installed hashicorp/azurerm v3.74.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
bash-3.2$ terraform apply
data.azurerm_resource_group.test_rg_01: Reading...
data.azurerm_resource_group.test_rg_01: Read complete after 1s [id=/subscriptions/xxx/resourceGroups/tf-provider-test-5000]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_data_protection_backup_vault.test_bv_01 will be created
  + resource "azurerm_data_protection_backup_vault" "test_bv_01" {
      + datastore_type      = "VaultStore"
      + id                  = (known after apply)
      + location            = "northeurope"
      + name                = "test-backup-vault-01"
      + redundancy          = "GeoRedundant"
      + resource_group_name = "tf-provider-test-5000"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azurerm_data_protection_backup_vault.test_bv_01: Creating...
azurerm_data_protection_backup_vault.test_bv_01: Still creating... [10s elapsed]
azurerm_data_protection_backup_vault.test_bv_01: Still creating... [20s elapsed]
azurerm_data_protection_backup_vault.test_bv_01: Creation complete after 24s [id=/subscriptions/xxx/resourceGroups/tf-provider-test-5000/providers/Microsoft.DataProtection/backupVaults/test-backup-vault-01]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
bash-3.2$ 
```

### Expected Behaviour

When creating a backup vault resource "azurerm_data_protection_backup_vault", terraform (azurerm) provider should have argument soft delete option to turn it on/off.

Instead when creating the same on portal, it has soft delete enabled by default.

### Actual Behaviour

Backup vault has soft delete disabled post Terraform creation.

### Steps to Reproduce

1. Run terraform apply (on the Terraform Configuration file provided above)
2. Go to azure portal and validate backup vault properties Soft delete (Preview), and it's disabled.
3. There is no option in Terraform resource to update this property.

Note: The issue has been observed across the following versions:
Terraform Version: 1.5.7
AzureRM Provider Version: 3.74.0

Terraform Debug message is attached here
https://gist.github.com/Peter0401/fde7aee665616c51fe95878d70cf9119

**Comment by rcskosir** [comment]
@Peter0401 Thank you for taking the time to open this feature request!

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #24775. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
